samples/nrf9160/serial_lte_modem/src/slm_at_gps.c

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <logging/log.h>
+LOG_MODULE_REGISTER(gps, CONFIG_SLM_LOG_LEVEL);
+
+#include <zephyr.h>
+#include <stdio.h>
+#include <net/socket.h>
+#include <nrf_socket.h>
+
+#include "slm_at_gps.h"
+
+#define FIX(_flag)       (_flag & NRF_GNSS_PVT_FLAG_FIX_VALID_BIT)
+#define UNHEALTHY(_flag) (_flag & NRF_GNSS_SV_FLAG_UNHEALTHY)
+
+#define INVALID_SOCKET	-1
+
+/**@brief List of supported AT commands. */
+enum slm_gps_mode {
+  GPS_MODE_STANDALONE,
+  GPS_MODE_PSM,
+  GPS_MODE_EDRX,
+  GPS_MODE_AGPS
+};
+
+/**@brief List of supported AT commands. */
+enum slm_gps_at_cmd_type {
+  AT_GPSRUN,
+  AT_GPS_MAX
+};
+
+/** forward declaration of cmd handlers **/
+static int handle_at_gpsrun(const char *at_cmd_param);
+
+/**@brief SLM AT Command list type. */
+static slm_at_cmd_list_t m_at_list [AT_GPS_MAX] = {
+	{AT_GPSRUN, "AT#XGPSRUN", "at#xgpsrun", handle_at_gpsrun},
+};
+
+static struct gps_client {
+	int sock; /* Socket descriptor. */
+	u16_t mask; /* NMEA mask */
+	bool running; /* GPS running status */
+	bool has_fix; /* At least one fix is got */
+	slm_at_callback_t callback;
+} client;
+
+static nrf_gnss_data_frame_t gps_data;
+static char buf[64];
+
+#define THREAD_STACK_SIZE	KB(1)
+#define THREAD_PRIORITY		K_PRIO_PREEMPT(7)
+
+static struct k_thread gps_thread;
+static k_tid_t gps_thread_id;
+static K_THREAD_STACK_DEFINE(gps_thread_stack, THREAD_STACK_SIZE);
+
+/* global variable defined in different files */
+extern struct at_param_list m_param_list;
+
+static void gps_satellite_stats(void)
+{
+	static u8_t last_tracked;
+	u8_t tracked = 0;
+	u8_t in_fix = 0;
+	u8_t unhealthy = 0;
+
+	if (gps_data.data_id != NRF_GNSS_PVT_DATA_ID)
+		return;
```

Comment:
Coding guideline requires surroding brackets `{ }` even for oneliners. Here and below.

Reply:
Updated